### PR TITLE
Remove monologue agent

### PR DESCRIPTION
**Pull request by neubig**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

`MonologueAgent` was the agent that started OpenDevin, but it is no longer used by default or to achieve any reasonable results. This PR removes it to reduce our maintenance burden.

Farewell `MonologueAgent`, it was nice to know you!

### CODE REVIEW

**Review by tobitege — APPROVED** [review]
LGTM.
Just wondering about the planner getting a response parser now?

File: agenthub/planner_agent/response_parser.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from opendevin.events.serialization.action import action_from_dict
 
 
-class MonologueResponseParser(ResponseParser):
+class PlannerResponseParser(ResponseParser):
```
[the remark this answers is not in the document]
Comment (neubig) [reply agenthub/planner_agent/response_parser.py:9]: Yes, previously `PlannerAgent` was using the response parser from `MonologueAgent` directly by importing it (maybe not the best design?). So I moved it over to the planner directory and renamed it.

[the remark this answers is not in the document]
Comment (tobitege) [reply agenthub/planner_agent/response_parser.py:9]: Oh, that makes total sense, I didn't look at that part.

[the remark this answers is not in the document]
Comment (tobitege) [reply agenthub/planner_agent/response_parser.py:9]: But the planner_agent's `prompt.py` also has a `parse_response` method. 🤔

**Pull request merged**